### PR TITLE
Added Base DatasetTransformer

### PR DESCRIPTION
**Pull request by Saurav-D**
# Peer Review Information

Added BaseTransformer base class for dataset transformer. 
Update COCO-Instances transformer to inherit base transformer.
Update COCO-Keypoints transformer to inherit base transformer.

# Pull Request Check List

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: datasetinsights/datasets/transformers/base.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        )
+
+    return transformer(**kwargs)
+
+
+class DatasetTransformer(ABC):
+    """ Base class for all ddataset transformer.
+    """
+
+    def __init__(self, **kwargs):
```
Comment (adason) [inline datasetinsights/datasets/transformers/base.py]: Do we need the __init__ method here as part of the abstract class?

File: datasetinsights/datasets/transformers/base.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+_registry = {}
```
Comment (adason) [inline datasetinsights/datasets/transformers/base.py]: Can this registry be a class variable of `DatasetTransformer`?

**Review by adason — DISMISSED** [review]

File: datasetinsights/datasets/transformers/base.py (hunk 2 above)
Comment (Saurav-D) [reply datasetinsights/datasets/transformers/base.py]: Yes, done.

File: datasetinsights/datasets/transformers/base.py (hunk 1 above)
Comment (Saurav-D) [reply datasetinsights/datasets/transformers/base.py]: Removed.

**Review by adason — APPROVED** [review]

**Review by salehe-ee — APPROVED** [review]
Changes look good to me.

**Pull request merged**